lib/segment/src/spaces/simple.rs

```diff
@@ -74,23 +60,111 @@ impl Metric for CosineMetric {
     }
 
     fn similarity(&self, v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
-        v1.iter().zip(v2).map(|(a, b)| a * b).sum()
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+        {
+            if is_x86_feature_detected!("avx2") {
```

Comment:
is it possible to also get some performance boost out of SSE* SIMD operations on older architectures?

Reply:
It is possible to add AVX1 support. 10-years old processors have AVX2 instructions. Do we need support SIMD for older CPUs?